File: competitive_programming/leetcode/06_javascript/191_number_of_1_bits.js
function numberOf1Bits(n) {
  let value = n >>> 0;
  let count = 0;
  while (value) {
    count += value & 1;
    value >>>= 1;
  }
  return count;
}

if (require.main === module) {
  if (numberOf1Bits(11) !== 3 || numberOf1Bits(128) !== 1 || numberOf1Bits(4294967293) !== 31) throw new Error("number of 1 bits failed");
  console.log("191 number of 1 bits ok");
}

File: competitive_programming/leetcode/11_golang/191_number_of_1_bits.go
package main

import (
	"fmt"
	"math/bits"
)

func numberOf1Bits(n uint) int {
	return bits.OnesCount(n)
}

func main() {
	if numberOf1Bits(11) != 3 || numberOf1Bits(128) != 1 || numberOf1Bits(4294967293) != 31 {
		panic("number of 1 bits failed")
	}
	fmt.Println("191 number of 1 bits ok")
}
